### PR TITLE
Added "native" kwarg to add_XXX_args. Closes #3669.

### LINKED ISSUE #3669: add_project_link_arguments adds arguments to native compiler
I am using `add_project_link_arguments` to set linker arguments but those end up applied to the native compiler not the cross-compiler.

This is IMO a bit unexpected but I guess changing this is not really possible without breaking things.
So maybe a `native` kwarg can be added, so that it can be chosen to which compiler the flags will be added.

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: docs/markdown/snippets/native_args.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## Projects args can be set separately for cross and native builds (potentially breaking change)
+
+It has been a longstanding bug (or let's call it a "delayed bug fix")
+that if yodo this:
```
Comment (ePirat) [inline docs/markdown/snippets/native_args.md]: Nitpick: Typo `yodo`.

File: test cases/common/23 global arg/prog.c [hunk 2]
```diff
@@ -10,6 +10,29 @@
...
+
+#ifdef GLOBAL_CROSS
+#ifndef ARG_CROSS
+#error "Global is cross but arg_cross is not set."
+#endif
+#else
+#ifdef ARG_CROSS
+#error "Global is cross but arg_native is set."
+#endif
+#endif
```
Comment (nirbheek) [inline test cases/common/23 global arg/prog.c]: Would be good to add indentation here so that the if/else/endif nesting is clearer.

File: docs/markdown/Reference-manual.md [hunk 3]
```diff
@@ -13,13 +13,20 @@ afterwards](#returned-objects).
...
+applied to. If a list of languages is given, the arguments are added
+to each of the corresponding compiler command lines. Note that there
+is no way to remove an argument set in this way. If you have an
+argument that is only used in a subset of targets, you have to specify
+it in per-target flags.
+
+- `native` is a boolean specifying whether the arguments should be
+  applied to the native or cross compilation. If omitted, the flags
+  are added to native targets if compiling natively and cross targets
+  if cross compiling. Available since 0.48.0
```
Comment (nirbheek) [inline docs/markdown/Reference-manual.md]: This para is a bit confusing. I would reword it as: 

```markdown
- `native` when set to `true`, the arguments will be added only for 
   native compilations when cross-compiling. If omitted or `false`, 
   the arguments are added to native targets if compiling natively
   and cross targets if cross compiling. Available since 0.48.0
```

Comment (jpakkane) [reply docs/markdown/Reference-manual.md]: That's not the same wording. If set to `false` then they _won't_ be added to native targets ever. The table goes (or should go at least) like this:

```
true => only to native builds
false => only to cross builds
omitted => native builds when native compiling, cross builds _only_ when cross building
```

Comment (nirbheek) [reply docs/markdown/Reference-manual.md]: Perhaps it's better to put it in a table form? Should be easier to parse than a paragraph.

File: docs/markdown/snippets/native_args.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+```
+## Added to native builds when compiling natively and to cross
+## compilations when doing cross compiles.
+add_project_arguments(...)
+
+## Added only to native compilations, not used in cross compilations.
+add_project_arguments(..., native : true)
+
+## Added only to corss compilations, not used in native compilations.
```
Comment (Ericson2314) [inline docs/markdown/snippets/native_args.md]: typo `corss`

**Comment by Ericson2314** [comment]
Per #3972 and #3969, what you *actually* want is not `native` vs `cross`, but `build` vs `host`. I.e. some flags are for code-gen tools, some flags are for the final installed artifact, and rarely are flags for both. Crucially: this "rarely" is true whether or not we are cross-compiling.

You could instead have have:
 - no extra arg continue to apply to both for 100% backwards compat
 - `target: "build"`: Apply only to code-gen stuff, i.e. builds that stay native even when cross compiling
 - `target: "host"`: Apply only to installable stuff, i.e. builds that become cross when cross compiling

`target: "build"` matches the no-arg default this PR currently has, `target: "host"` has the opposite effect.

**Comment by jpakkane** [comment]
Should work now.

**Comment by jpakkane** [comment]
We have `native` in all other functions like this, for consistency we should use the same here. Having the same thing called different things in different places is bad UX.
